hitrace/src/lib.rs

```diff
@@ -70,6 +70,103 @@ pub fn finish_trace() {
     finish_trace_()
 }
 
+/// Wrapper function for `OH_HiTrace_CountTrace` with a CStr name parameter
+#[cfg(all(target_env = "ohos", not(feature = "max_level_off")))]
+fn trace_metric_cstr(name: &CStr, count: i64) {
+    unsafe {
+        hitrace_sys::OH_HiTrace_CountTrace(name.as_ptr(), count);
+    }
+}
+
+#[cfg(any(not(target_env = "ohos"), feature = "max_level_off"))]
+fn trace_metric_cstr(_: &CStr, _: i64) {}
+
+/// A function to log a count trace event with a name and any integer count that implements `Into<i64>`.
+fn trace_metric<T: AsRef<CStr>, C: Into<i64>>(name: &T, count: C) {
+    trace_metric_cstr(name.as_ref(), count.into());
+}
+
+/// Logs a count trace event with a `&str` name and an integer count that can be converted to `i64`.
+///
+/// # Arguments
+/// * `name` - The name of the event.
+/// * `count` - The integer count, convertible to `i64`.
+///
+/// # Panics
+/// Panics if `name` contains a null byte.
+pub fn trace_metric_str<C: Into<i64>>(name: &str, count: C) {
+    let c_string = CString::new(name).expect("Failed to convert to CString");
+    trace_metric(&c_string, count.into());
+}
+
+/// Logs a count trace event with a name and an integer count, using saturating conversion to `i64`.
+///
+/// # Arguments
+/// * `name` - The name of the event (as `CStr` or `CString`).
+/// * `count` - The integer count, implementing `SaturatingIntoI64`.
+pub fn trace_metric_saturating<T: AsRef<CStr>>(name: &T, count: impl SaturatingIntoI64) {
+    trace_metric_cstr(name.as_ref(), count.saturating_into());
+}
+
+/// Logs a count trace event with a `&str` name and an integer count, using saturating conversion to `i64`.
+///
+/// # Arguments
+/// * `name` - The name of the event.
+/// * `count` - The integer count, implementing `SaturatingIntoI64`.
+///
+/// # Panics
+/// Panics if `name` contains a null byte.
+pub fn trace_metric_saturating_str(name: &str, count: impl SaturatingIntoI64) {
```

Comment:
@jschwe  I renamed it to `trace_metric` and created saturating and made interface accept count `Into<i64>`